shelley/chain-and-ledger/executable-spec/src/Delegation/Certificates.hs

```diff
@@ -119,6 +120,11 @@ newtype PoolDistr hashAlgo dsignAlgo vrfAlgo =
   PoolDistr (Map (KeyHash hashAlgo dsignAlgo) (Rational, Hash hashAlgo (VerKeyVRF vrfAlgo)))
   deriving (Show, Eq, NoUnexpectedThunks, Relation)
 
-isInstantaneousRewards :: (DCert hashAlgo dsignAlgo vrfAlgo) -> Bool
+isInstantaneousRewards :: DCert hashAlgo dsignAlgo vrfAlgo -> Bool
 isInstantaneousRewards (InstantaneousRewards _) = True
 isInstantaneousRewards _                        = False
+
+hasWitness :: DCert hashAlgo dsignAlgo vrfAlgo -> Bool
```

Comment:
counting beans here, but wouldn't it be better to use something like `requiresVKeyWitness` ? It would also be good to add documentation for this function.

Reply:
ah, sure, `requiresVKeyWitness` is a better name. I'll change it and add a docstring.